controllers/storagecluster/cephcluster.go

```diff
@@ -175,11 +175,13 @@ func (obj *ocsCephCluster) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.
 			}
 			if kmsConfigMap != nil {
 				if err = reachKMSProvider(kmsConfigMap); err != nil {
```

Comment:
As you are any way making changes to this file a few suggestions,
Why add the `Status.KMSServerConnection` details only when there is an error?  A successful connection should populate the address field. (nobody to blame but me =))
Can we populate `Status.KMSServerConnection.KMSServerAddress` before the error and add any error details to `KMSServerConnection.KMSServerConnectionError` if there is an error.

Otherwise `KMSServerAddress` field will only be populated when there is an error...

```suggestion
				if kmsConfigMap.Data["KMS_PROVIDER"] == "vault" {
					sc.Status.KMSServerConnection.KMSServerAddress = kmsConfigMap.Data["VAULT_ADDR"]
				}
				if err = reachKMSProvider(kmsConfigMap); err != nil {
					sc.Status.KMSServerConnection.KMSServerConnectionError = err.Error()
					r.Log.Error(err, "Address provided in KMS ConfigMap is not reachable.", "KMSConfigMap", klog.KRef(kmsConfigMap.Namespace, kmsConfigMap.Name))
```

Reply:
Sure, updated the code as per the suggestion.